### ISSUE TITLE
Changes to pipeline.state within backend is not considered for postprocessing

### ISSUE THREAD

**Issue by Mat0vu** [SigmaHQ/pySigma#311]
Hi,

When experimenting with the `set_state` transformation and a postprocessing pipeline for EQL-ndjson I discovered some issue with the handling of the pipeline `state` variable: 
I have a set_state transformation to assign indizes in which the search shall be executed for different log sources like this:
```yaml
    key: index
    val: "*winlog*"
    rule_conditions:
      - type: logsource
        product: windows
```

Because Elasticsearch demands a list of indizes for the import of EQL Detection Rules, I wanted to ensure that the value set with the `set_state` transformation is a list. Therefore I added this into the `finalize_query` function of the EQL Backend (https://github.com/SigmaHQ/pySigma-backend-elasticsearch/pull/104).

```python
        index_state = state.processing_state.get("index")
        if not index_state:
            index_state = self.index_names
        if not isinstance(index_state, list):
            index_state = [index_state]
        # Save the processed index back to the processing state
        state.processing_state["index"] = index_state
        return super().finalize_query(rule, query, index, state, output_format)
```

However, then I discovered that even though passing the Indizes as a list in the `state` variable, in the `base.py` class when `self.last_processing_pipeline.postprocess_query(rule, backend_query)` is called, `self.last_processing_pipeline.state` is still containing `{"index": "*winlog*"}` (probably just representing the yaml definition of the pipeline). The problem is that this is then also used by the jinja template and not the adjusted `state` which contains the corrected value (see screenshot)

![image](https://github.com/user-attachments/assets/70606864-675b-499b-9efe-2f9aca25f625)

Obviously this could be easily avoided by specifying the `val` values of the transformation as a list in this context but I expected that by adjusting the pipeline state variable, this would also reflect in the result of the postprocessing. 
What do you think about this, should this be changed or is this the desired behaviour?

**Comment by thomaspatzke** [comment]
Ok, you're getting really into the deep secrets of pySigma 😉

Let me explain the situation. There are two states:

1. The *pipeline state* stored in `self.last_processing_pipeline.state`. This pipeline with its state is reconstructed [for each converted rule](https://github.com/SigmaHQ/pySigma/blob/30660ed4ea048aca1f5b3e5e2629991f0da78212/sigma/conversion/base.py#L183). This state is where the transformations operate on. Specifically here, where the state is set and the Jinja template take their variables from. This one is intended to be set by the user via processing pipelines to control the result.
2. The *conversion state*, that is [initialized here](https://github.com/SigmaHQ/pySigma/blob/30660ed4ea048aca1f5b3e5e2629991f0da78212/sigma/conversion/base.py#L197) with the state of the pipeline. This state is mainly intended to be used only internally in the backend to being able to pass state information while the parse tree is traversed.

Both states differ slightly in their lifetime: the pipeline state exists for the whole rule while the conversion state exists separately for each condition (one Sigma rule can have multiple conditions). Very similar things for slightly different purposes.

Basically your array enforcement code sets the conversion state but the query postprocessing state uses the pipeline state. Setting the pipeline state in your code should solve the issue:
```python
self.last_processing_pipeline.state["index"] = index_state
```
